rfd/0093-offline-access.md

```diff
@@ -0,0 +1,190 @@
+---
+authors: Tim Ross (tim.ross@goteleport.com)
+state: draft
+---
+
+# RFD 0093 - Robust Access to Nodes
+
+
+# Required Approvers
+
+* Engineering @zmb3 && (fspmarshall || codingllama)
+
+## What
+
+Ensure that users can access nodes via `tsh ssh` when there is intermittent or no connectivity to Auth.
+
+## Why
+
+Connectivity to the Auth server is a single point of failure which can result in not being able to access
+a resource. If a user already has already obtained a valid certificate (via `tsh login`) there are certain
+scenarios when a connection to Auth should not required to gain access to a node. This would reduce the burden
+on cluster admins when performing an Auth upgrade. On top of the potential downtime during the upgrade, any issues 
+found as a result of the upgrade require a rollback, which can further extend downtime and lack of connectivity.
+
+## Details
+
+### Auth Connection Required 
+
+The following cluster settings or operations **must** have a connection to Auth server to establish a connection to a node:
+
+#### Label based tsh ssh
+
+```bash
+tsh ssh user@"foo=bar"
+```
+
+For the above command to work, `tsh` needs to be able to resolve all matching nodes which have the label
+`foo=bar`. To determine which nodes match the provided labels `tsh` issues a `proto.AuthService/ListResources`
+request to the Auth server which returns the matching node set. Without a connection to Auth node resolution
+is just not possible.
+
+
+#### Sync Recording modes
+
+Both `proxy-sync` and `node-sync` recording modes require a persistent connection to Auth to upload session recordings. As a result, even if `tsh ssh` doesn't require a connection to Auth, the session does and as a result the node will prevent a session from starting.
```

Comment:
What is the failure mode here? An initial connection is made, but then an error is thrown? Or do we block the initial connection attempt?